CODE_OF_CONDUCT.rst

```diff
@@ -0,0 +1,154 @@
+=====================
+Xonsh Xode of Xonduct
+=====================
+
+Xonsh is an engaged and respectful community made up of people from all over the
+world. Your involvement helps us to expand and strengthen our community and to
+further the progress of ``xonsh`` and related tooling.
+
+To quote one Anthony Scopatz:
+
+    It turns out shells have a lot of tricky edge cases.
+
+Add to this that shells are personal tools that invite customization. This, in
+turn, invites lots of deeply-held convictions and strong opinions.  
+
+These are all *good things*!
+
+This implies diversity of ideas and perspectives on often complex
+problems. Disagreement and healthy discussion of conflicting viewpoints is
+welcome: the best solutions to hard problems rarely come from a single angle.
+But disagreement is not an excuse for aggression: humans tend to take
+disagreement personally and easily drift into behavior that ultimately degrades
+a community. This is particularly acute with online communication across
+language and cultural gaps, where many cues of human behavior are unavailable.
+We are outlining here a set of principles and processes to support a
+healthy community in the face of these challenges.
+
+Fundamentally, we are committed to fostering a productive, harassment-free
+environment for everyone. Rather than considering this code an exhaustive list
+of things that you can’t do, take it in the spirit it is intended - a guide to
+make it easier to enrich all of us and the communities in which we participate.
+
+Importantly: as a member of our community, *you are also a steward of these
+values*.  Not all problems need to be resolved via formal processes, and often
+a quick, friendly but clear word on an online forum or in person can help
+resolve a misunderstanding and de-escalate things.
+
+This code applies equally to founders, developers, mentors and new community
+members, in all spaces managed by Xonsh. This includes all repositories in the
+``xonsh`` GitHub organization, the mailing list, all chat rooms (Gitter and
+IRC), and any other forums created by the project team. In addition, violations
+of this code outside these spaces may affect a person's ability to
+participate within them.
+
+.. note:: It is the strong opinion of the Xonsh developers that the maintainers
```

Comment:
Ooph, hot take

Reply:
Happy for any language / policy tweaks that folks want to suggest!